src/cryptography/x509/verification.py

```diff
@@ -0,0 +1,93 @@
+# This file is dual licensed under the terms of the Apache License, Version
+# 2.0, and the BSD License. See the LICENSE file in the root of this repository
+# for complete details.
+
+from __future__ import absolute_import, division, print_function
+
+from cryptography.hazmat.primitives.asymmetric import ec, padding, rsa
+from cryptography.x509 import Certificate
+from cryptography.x509.oid import ExtensionOID
+
+
+class InvalidCertificate(Exception):
+    pass
+
+
+class InvalidSigningCertificate(Exception):
+    pass
+
+
+def _can_sign_certificates(certificate):
+    basic_constraints = certificate.extensions.get_extension_for_oid(
+        ExtensionOID.BASIC_CONSTRAINTS).value
+    key_usage = certificate.extensions.get_extension_for_oid(
+        ExtensionOID.KEY_USAGE).value
+
+    if not basic_constraints.ca:
+        raise InvalidSigningCertificate(
+            "The certificate is not marked as a CA in its BasicConstraints "
+            "extension."
+        )
+    elif not key_usage.key_cert_sign:
+        raise InvalidSigningCertificate(
+            "The certificate public key is not marked for verifying "
+            "certificates in its KeyUsage extension."
+        )
+    else:
+        return True
+
+
+def _is_issuing_certificate(issuing_certificate, issued_certificate):
+        return (issuing_certificate.subject == issued_certificate.issuer)
+
+
+class CertificateVerificationContext(object):
```

Comment:
Right now the context only examines the Basic Constraints and Key Usage extensions for fields required for CA use. Other extensions are ignored.